docs/manual/lepton-netlist-backends.texi

```diff
@@ -102,3 +68,326 @@ Backends for automation.
+extension @file{.scm}.  A backend of any type should define a main
+function named the same as the backend itself.  Please note that the
+name of a backend is not the same as of its file name.  Subtle
+differences in naming will be revealed in the next sections.
+
+In both cases @netlist{} loads and evaluates the main backend function
+at a certain stage of its execution.
+
+
+@node Legacy backends
+@subsubsection Legacy backends
+@cindex legacy backends
+@cindex lepton-netlist legacy backends
+
+Legacy backend system has been inherited from gEDA @command{gnetlist}.
+
+In order @netlist{} to understand a file is a legacy backend, it must
+meet the following requirements:
+@itemize
+@item
+Its filename must have the @file{gnet-} prefix.
+@item
+The filename must have the extension @file{.scm}.
+@item
+It should export a function named the same as backend itself.
+@end itemize
+
+The name of the backend is defined by removing the prefix @file{gnet-}
+and the suffix @file{.scm} from its file basename.  For example, if
+backend file name is @file{gnet-custom-backend.scm}, its name is
+@code{custom-backend} and its code has to contain a definition of a
+function named @code{custom-backend}.  The function must take one
+argument which is the output file name.  Really, it is not necessary
+to use this argument in the code as @netlist{} automatically redirects
+the standard output of the function into a file if it is specified by
+its command-line arguments.
+
+
+@node Module backends
+@subsubsection Module backends
+@cindex module backends
+@cindex lepton-netlist module backends
+
+Unlike legacy backends, a module backend is simply a Scheme module.
+In order @netlist{} to understand a file is a module backend, it must
+meet the following requirements:
+
+@itemize
+@item
+The filename must have the extension @file{.scm}.
+@item
+It must define the module in the @samp{backend} namespace and export
+the same named function.
+@item
+It must live in one of Lepton's user or system data directories.
+@end itemize
+
+For example, the name of your backend is @file{my-backend.scm}.  In
+the code, you have to add module definition as follows:
+
+@lisp
+(define-module (backend my-backend)
+  #:export (my-backend))
+@end lisp
+
+The exported function has take no arguments.  What it outputs to the
+standard output port is redirected by @netlist{} to the specified
+output file or standard output, or to the default output name if no
+output file name is specified.
+
+The function may be defined as follows:
+@lisp
+(define (my-backend)
+  (some-code-follows) ...)
+@end lisp
+
+To make the function take an arbitrary list of arguments, it can be
+defined as follows:
+@lisp
+(define (my-backend . args)
+  (some-code-that-use-args args) ...)
+@end lisp
+
+So, if you want the backend to be available system-wide, put it in the
+@file{scheme/backend} subdirectory of one of the directories reported
+by the function @code{sys-data-dirs}, e.g. @file{/etc/lepton-eda}.  To
```

Comment:
`/usr/share/lepton-eda` would be more appropriate in this example.
Directories under `/etc/` are printed rather by `sys-config-dirs()`.